From 73f7d86b37a0b0d4e731ef62047f2ad2ba2a111f Mon Sep 17 00:00:00 2001
From: Tatyana Nikolova <tatyana.e.nikolova@intel.com>
Date: Mon, 13 Mar 2023 21:40:04 -0500
Subject: [PATCH] perftest: Add Intel device names and inline data sizes

Add two Intel devices and their corresponding
inline data sizes.

Signed-off-by: Tatyana Nikolova <tatyana.e.nikolova@intel.com>
Signed-off-by: Kamal Heib <kheib@redhat.com>
---
 src/perftest_parameters.c | 12 ++++++++++--
 src/perftest_parameters.h |  3 ++-
 2 files changed, 12 insertions(+), 3 deletions(-)

diff --git a/src/perftest_parameters.c b/src/perftest_parameters.c
index a587df3beb06..6b87621cea0e 100755
--- a/src/perftest_parameters.c
+++ b/src/perftest_parameters.c
@@ -1794,7 +1794,11 @@ enum ctx_device ib_dev_name(struct ibv_context *context)
 		If you want Inline support in other vendor devices, please send patch to gilr@dev.mellanox.co.il
 		*/
 	} else if (attr.vendor_id == 0x8086) {
-		dev_fname = INTEL_ALL;
+		switch (attr.vendor_part_id) {
+			case 14289 : dev_fname = INTEL_GEN1; break;
+			case 5522  : dev_fname = INTEL_GEN2; break;
+			default    : dev_fname = INTEL_GEN2; break;
+		}
 	} else {
 
 		//coverity[uninit_use]
@@ -2080,7 +2084,7 @@ static void ctx_set_max_inline(struct ibv_context *context,struct perftest_param
 			return;
 		}
 		#endif
-		if (user_param->tst ==LAT) {
+		if (user_param->tst == LAT) {
 			switch(user_param->verb) {
 				case WRITE: user_param->inline_size = (user_param->connection_type == DC)? DEF_INLINE_DC : DEF_INLINE_WRITE; break;
 				case SEND : user_param->inline_size = (user_param->connection_type == DC)? DEF_INLINE_DC : (user_param->connection_type == UD)? DEF_INLINE_SEND_UD :
@@ -2097,6 +2101,10 @@ static void ctx_set_max_inline(struct ibv_context *context,struct perftest_param
 				user_param->inline_size = 96;
 			else if (current_dev == HNS)
 				user_param->inline_size = 32;
+			else if (current_dev == INTEL_GEN1)
+				user_param->inline_size = 48;
+			else if (current_dev == INTEL_GEN2)
+				user_param->inline_size = 101;
 
 		} else {
 			user_param->inline_size = 0;
diff --git a/src/perftest_parameters.h b/src/perftest_parameters.h
index f2080fad0b6d..2f846f78f1c2 100755
--- a/src/perftest_parameters.h
+++ b/src/perftest_parameters.h
@@ -373,7 +373,7 @@ enum ctx_device {
 	MLX5GENVF		= 19,
 	BLUEFIELD		= 20,
 	BLUEFIELD2		= 21,
-	INTEL_ALL		= 22,
+	INTEL_GEN1		= 22,
 	NETXTREME		= 23,
 	EFA			= 24,
 	CONNECTX6LX		= 25,
@@ -383,6 +383,7 @@ enum ctx_device {
 	ERDMA			= 29,
 	HNS			= 30,
 	CONNECTX8		= 31,
+	INTEL_GEN2		= 32,
 };
 
 /* Units for rate limiter */
-- 
2.41.0

